### DISCUSSION TITLE
How to show a dialog on a navigation view item select?

### DISCUSSION THREAD

**Discussion by ClaudiuHBann** [amwx/FluentAvalonia#501] [Q&A]
I have an "About" item in the navigation view and i don't have that much to show, so making a separate page is too much and i want to show a small dialog, how can i achieve this?

**Comment by timunie** [comment]
check https://github.com/AvaloniaUI/Avalonia.Samples?tab=readme-ov-file#view-interaction-samples

**Reply by ClaudiuHBann** [reply]
I am sorry, but I can't see how this can help me...

**Reply by timunie** [reply]
from your VM you can show a dialog if selected new item is AboutVM. And after "OK" you can navigate back to whatever was selected before.

**Reply by ClaudiuHBann** [reply]
this is the approach I have right now, but I see this as a workaround, I thought there is a "perfect" solution

**Reply by ClaudiuHBann** [reply]
is there an event for the selection changing of the navigation view, and if it is, can i cancel the selection?

**Reply by timunie** [reply]
https://github.com/amwx/FluentAvalonia/tree/master/src/FluentAvalonia/UI/Controls/NavigationView/EventArgs

maybe ItemInvoked?

**Comment by amwx** [comment]
On the NavigationViewItem you want to use to open the dialog, set the property `SelectsOnInvoked` to false, then handle the `ItemInvoked` event on the NavigationView and check for your item. If its invoked, open your dialog.
